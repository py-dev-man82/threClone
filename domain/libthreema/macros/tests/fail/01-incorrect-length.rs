use libthreema_macros::concat_fixed_bytes;

fn main() {
    let a = [1u8; 4];
    let b = [2u8; 3];
    let c = [3u8; 3];
    let d = [4u16; 2];

    let result: [u8; 8] = concat_fixed_bytes!(a, b); // yields 7 bytes, not 8
    let result: [u8; 8] = concat_fixed_bytes!(a, b, c); // yields 10 bytes, not 8
}
