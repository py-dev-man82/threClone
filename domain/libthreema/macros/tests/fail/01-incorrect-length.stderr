error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
 --> tests/fail/01-incorrect-length.rs:9:27
  |
9 |     let result: [u8; 8] = concat_fixed_bytes!(a, b); // yields 7 bytes, not 8
  |                           ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: source type: `main::ConcatenatedArrays<4, 3>` (56 bits)
  = note: target type: `[u8; 8]` (64 bits)
  = note: this error originates in the macro `concat_fixed_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
  --> tests/fail/01-incorrect-length.rs:10:27
   |
10 |     let result: [u8; 8] = concat_fixed_bytes!(a, b, c); // yields 10 bytes, not 8
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: source type: `main::ConcatenatedArrays<4, 3, 3>` (80 bits)
   = note: target type: `[u8; 8]` (64 bits)
   = note: this error originates in the macro `concat_fixed_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)
